src/locks/locks/blocklocks/BlockLock.ts

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { Location, Material, Rotation } from 'org.bukkit';
+import { Block, BlockFace } from 'org.bukkit.block';
+import { Directional } from 'org.bukkit.block.data';
+import { ItemFrame, Player } from 'org.bukkit.entity';
+import { LockCustomItem, LockDataType, LockItem } from '../lock-items';
+
+export interface BlockLockProps {
+  itemFrame: ItemFrame;
+  lockData: LockDataType;
+  lockCustomItem: LockCustomItem;
+  block: Block;
+}
+
+export abstract class BlockLock {
```

Comment:
Public member access is the default in TypeScript, no need to specify it explicitly.